#!/usr/bin/env python
PACKAGE = "gcop_ctrl"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#dilation of the obstacle
gen.add("dyn_obs_dilation_m",     double_t, 0, "The amount by which obstacles are expanded in meters ", 1, 0, 5)
enum_dilation_type = gen.enum([gen.const("MORPH_RECT",    int_t, 0, "rect"),  
                               gen.const("MORPH_CROSS",   int_t, 1, "cross"),
                               gen.const("MORPH_ELLIPSE", int_t, 2, "ellipse")],"An enum to set dilation type")
gen.add("dyn_dilation_type"  ,    int_t, 0, "Select the dilation type"  , 2, 0, 2, edit_method=enum_dilation_type)


gen.add("dyn_debug_on",    bool_t,   0, "enable debug", False)
gen.add("dyn_plan_dsl_global",    bool_t,   0, "plan the path", False)
gen.add("dyn_smooth_global_path", bool_t,   0, "fit a curve/spline to the global path", True)
gen.add("dyn_plan_ddp_local",     bool_t,   0, "plan the path", False)
gen.add("dyn_max_speed",           double_t, 0, "Max speed for the vehicle", 0.7, 0.5, 1.5 )
# 


exit(gen.generate(PACKAGE, "dstar_ddp_planner", "DstarDdpPlanner"))
